test/network/scenarios/common.js

```diff
@@ -17,6 +17,9 @@
 const childProcess = require('child_process');
 const utils = require('../utils');
 
+const NODE_READY_REGEX = /Finished sync/;
+const NODE_READY_TIMEOUT = 20000;
```

Comment:
the disadvantage of having the timeout as a way to check if the application starts within this time frame is that it will be indeterministic if the performance of the application degrades then the tests will fail and if the performance improves then still it has to wait 20 seconds, as I said before if we subscribe to `blockchainReady` event then regardless of the time the tests will be consistent.

Reply:
The timeout specifies the upper bound; there needs to be a timeout in case the node is taking way too long to launch; then it should fail the test with an error. It should be a large value though. I can make it higher. 20s is a bit short.

Reply:
I made it 40 seconds.

Reply:
I agree that it's not the best but we're running the node with PM2 so we can't listen to the node's events directly. I investigated using the HTTP API to check the sync status of the node but that would add a lot of complexity because of how we're running the nodes, it's difficult to associate them with a URL.